test/unit/math/prim/mat/prob/categorical_logit_test.cpp

```diff
@@ -79,3 +81,58 @@ TEST(ProbDistributionsCategoricalLogit, error) {
   ns[1] = 12;
   EXPECT_THROW(categorical_logit_log(ns, theta), std::domain_error);
 }
+
+TEST(ProbDistributionsCategoricalLogit, error_check) {
+  using stan::math::categorical_logit_rng;
+  boost::random::mt19937 rng;
+
+  Matrix<double, Dynamic, 1> beta(3);
+
+  beta << 1.0, 10.0, -10.0;
+
+  EXPECT_NO_THROW(categorical_logit_rng(beta, rng));
+
+  beta << -1e3, 1.1e3, 1e5;
```

Comment:
Should there be an `EXPECT_THROW` test for `beta` values with a range that can't be handled by the softmax? I think the answer is no and your implementation is fine (since underflow/overflow in generated quantities should be ok and we don't want to throw there much anyway, I'm just pointing out the concern since it would be nice to handle it somehow.